src/middleware/builtin.ts

```diff
@@ -146,15 +146,23 @@ export function matchConstraints(
 export function matchMessage(pattern: string | RegExp): Middleware<SlackEventMiddlewareArgs<'message'>> {
   return ({ message, context, next }) => {
     let tempMatches: RegExpExecArray | null;
+    const matches: string[] = [];
 
     // Filter out messages that don't contain the pattern
     if (typeof pattern === 'string') {
       if (message.text !== undefined && !message.text.includes(pattern)) {
         return;
       }
     } else {
-      if ((tempMatches = pattern.exec(message.text)) !== null) {
-        context['matches'] = tempMatches;
+      while ((tempMatches = pattern.exec(message.text)) !== null) {
+        matches.push(tempMatches[1]);
```

Comment:
isn't `tempMatches[1]` only the value that matches the first capture group? if there are several capture groups in the RegExp, then they would populate `tempMatches[2]` and beyond, and those would never be available to the user, right?

Reply:
Yes, you're right unfortunately. I was trying to find a way to avoid passing the whole RegExp exec result, but that may just be bad design.